corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -186,6 +188,48 @@ def get_http_response(self):
         return stream_response(open(self.get_filename(), 'r'), headers)
 
 
+class ObjectRestoreResponse(FileRestoreResponse):
+    """
+    Restore response powered by RiakCS backend
+    """
+    def __init__(self, username=None, items=False, domain=None):
+        super(ObjectRestoreResponse, self).__init__(username, items, domain)
+        self.object_store = ObjectStore(domain)
+
+    def finalize(self):
+        """
+        Creates the final file with start and ending tag
+        """
+        with open(self.get_filename(), 'w') as response:
+            # Add 1 to num_items to account for message element
+            items = self.items_template.format(self.num_items + 1) if self.items else ''
+            response.write(self.start_tag_template.format(
+                items=items,
+                username=self.username,
+                nature=ResponseNature.OTA_RESTORE_SUCCESS
+            ))
+
+            self.response_body.seek(0)
+            shutil.copyfileobj(self.response_body, response)
+
+            response.write(self.closing_tag)
+            content_length = os.path.getsize(self.get_filename())
+
+        self.object_store.set(
+            self.get_filename(),
+            open(self.get_filename(), 'rb'),
+            content_length,
+        )
```

Comment:
I think I saw something like that with the multi-part uploads: https://github.com/boto/boto3/blob/develop/tests/integration/test_s3.py#L193